packages/url_launcher/url_launcher_android/lib/url_launcher_android.dart

```diff
@@ -62,16 +60,57 @@ class UrlLauncherAndroid extends UrlLauncherPlatform {
     required Map<String, String> headers,
     String? webOnlyWindowName,
   }) async {
+    return launchUrl(
+        url,
+        LaunchOptions(
+            mode: useWebView
+                ? PreferredLaunchMode.inAppWebView
+                : PreferredLaunchMode.externalApplication,
+            webViewConfiguration: InAppWebViewConfiguration(
+                enableDomStorage: enableDomStorage,
+                enableJavaScript: enableJavaScript,
+                headers: headers)));
+  }
+
+  @override
+  Future<bool> launchUrl(String url, LaunchOptions options) async {
```

Comment:
The reason there's so much change here is that we never went back and implemented `launchUrl`, so we only had the legacy `launch` platform interface method. This adds the entire method so that we can support `inAppBrowserView`.

(That could be done as a prequel without the new mode, but it didn't seem worth splitting out just that small piece; if anyone would prefer that let me know.)